docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -415,67 +535,131 @@ class CustomerAccessFilter implements CustomCustomerAccessFilter
+        $this->customerAccessConfig = $customerAccessConfig;
+    }
+}
+```
+---
 
 ## PrivateApi:ObjectInitialization
 
-Business models on the project level should avoid initializing private API from the Core level.
+On the project level, business models must not initialize private API from the core level.
+
+### Example of error in the Evaluator output
+
+```bash
+------------------------------- -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+PrivateApi:ObjectInitialization Please avoid usage of Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface in Pyz\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter
+------------------------------- -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+```
 
-Example of code that causes an upgradability error:
+### Example of code that causes an upgradability error
 
-`CustomerAccessFilter` contains creating `CustomerAccessReader` in the `constructor` method.
+The `constructor` method in `CustomerAccessFilter` creates `CustomerAccessReader`.
 
 ```php
 ...
+use Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface;
+use Spryker\Zed\CustomerAccess\Persistence\CustomerAccessEntityManagerInterface;
+...
+
 class CustomerAccessFilter implements CustomCustomerAccessFilter
 {
+    ...
     public function __construct(
         CustomerAccessEntityManagerInterface $customerAccessEntityManager,
-        CustomerAccessReaderInterface $customerAccessReader,
         CustomerAccessFilterInterface $customerAccessFilter
     ) {
         parent::__construct($customerAccessEntityManager);
         $this->customerAccessReader = new CustomerAccessReader();
         $this->customerAccessFilter = $customerAccessFilter;
     }
+    ...
 }
 ```
 
-Example of related error in the Evaluator output:
+### Example of resolving the error
+
+Recommended: Apply the [development strategies](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html) that let you take updates safely.
 
-```bash
-------------------------------------------------------------------------------------------------------------------------
-Please avoid usage of Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface in Pyz\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter
-------------------------------------------------------------------------------------------------------------------------
-```
 
-Example of resolving the error by ?
+Not recommended:
+1. Register the missing extension point in [Spryker Ideas](https://spryker.ideas.aha.io/), so we add it in future.
+2. On the project level, create an entity manager with the methods from the core level.
```

Comment:
Please adjust the description of the step to match the examples.

Reply:
I don't know what to write here. It is just bad practice to create the objects of the class somewhere inside the code. Usage of the interface in the constructor relates to "Dependency Inversion Principle" from SOLID principles. 

So I have no idea how to change it in a bad way because the original code example is already bad, so that's why I don't know what to write here as "Not recommended".

Reply:
I copied step 2 from another rule. We just need to explain what we do in the following code snippet.